api/base/serializers.py

```diff
@@ -713,11 +714,28 @@ def to_representation(self, value):
             raise SkipField
 
         related_url = url['related']
+        related_path = urlparse(related_url).path
         related_meta = self.get_meta_information(self.related_meta, value)
         self_url = url['self']
         self_meta = self.get_meta_information(self.self_meta, value)
-        return format_relationship_links(related_url, self_url, related_meta, self_meta)
-
+        relationship = format_relationship_links(related_url, self_url, related_meta, self_meta)
+        if related_url and (len(related_path.split('/')) & 1) == 1:
+            resolved_url = resolve(related_path)
+            related_class = resolved_url.func.view_class
+            if not hasattr(related_class, 'LIST_FIELDS'):
```

Comment:
As discussed, better to check if the view derives from `mixins.mixins.RetrieveModelMixin` or doesn't derive from `mixins.ListModelMixin`

Reply:
Complete. I'll add a test and set ticket to CR.